app/frontend/src/lib/ui/components/filterGroup.js

```diff
@@ -46,14 +55,29 @@ export const init = (groupContainerSelector, removeButtonSelector, clearButtonSe
   }
 };
 
-export const displayOpenOrCloseText = (targetElement, expandedElements) => {
+export const displayOpenOrCloseText = (targetElement, expandedElements, maxElements) => {
   if (expandedElements === 0) {
     targetElement.innerText = OPEN_ALL_TEXT;
-  } else if (expandedElements > 0) {
+  } else if (expandedElements === maxElements) {
     targetElement.innerText = CLOSE_ALL_TEXT;
   }
 };
 
+export const addUpdateOpenOrCloseEvent = (buttonElement, openOrCloseAllSelector) => {
+  const openOrCloseAllElement = document.getElementById(openOrCloseAllSelector);
+  if (openOrCloseAllElement) {
+    buttonElement.addEventListener('click', (e) => {
+      let expandedElements = document.getElementsByClassName('govuk-accordion__section--expanded').length;
+      if (e.target.parentElement.parentElement.parentElement.classList.contains('govuk-accordion__section--expanded') && expandedElements !== 0) {
```

Comment:
use `closet` rather than `parentElement, parentElement...` for this as it makes the markup more flexible

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Element/closest and the polyfill is already used some where in here and hence already imported into this script